contracts/treasury/BondingCheckpointsVotes.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import "../Manager.sol";
+import "../bonding/IBondingCheckpoints.sol";
+import "./GovernorCountingOverridable.sol";
+
+/**
+ * @title Interface for BondingCheckpoints
+ */
+contract BondingCheckpointsVotes is Manager, IVotes {
+    // Indicates that the called function is not supported in this contract and should be performed through the
+    // BondingManager instead. This is mostly used for delegation methods, which must be bonds instead.
+    error MustCallBondingManager();
+
+    constructor(address _controller) Manager(_controller) {}
+
+    /**
+     * @notice Clock is set to match the current round, which is the checkpointing
+     *  method implemented here.
+     */
+    function clock() public view returns (uint48) {
+        return bondingCheckpoints().clock();
+    }
+
+    /**
+     * @notice Machine-readable description of the clock as specified in EIP-6372.
+     */
+    // solhint-disable-next-line func-name-mixedcase
+    function CLOCK_MODE() public view returns (string memory) {
+        return bondingCheckpoints().CLOCK_MODE();
+    }
+
+    /**
+     * @notice Returns the current amount of votes that `account` has.
+     */
+    function getVotes(address _account) external view returns (uint256) {
+        return getPastVotes(_account, clock());
+    }
+
+    /**
+     * @notice Returns the amount of votes that `account` had at a specific moment in the past. If the `clock()` is
+     * configured to use block numbers, this will return the value at the end of the corresponding block.
+     */
+    function getPastVotes(address _account, uint256 _round) public view returns (uint256) {
+        (uint256 amount, ) = bondingCheckpoints().getBondingStateAt(_account, _round);
+        return amount;
+    }
+
+    /**
+     * @notice Returns the total supply of votes available at a specific round in the past.
+     * @dev This value is the sum of all *active* stake, which is not necessarily the sum of all voting power.
+     * Bonded stake that is not part of the top 100 active transcoder set is still given voting power, but is not
+     * considered here.
+     */
+    function getPastTotalSupply(uint256 _round) external view returns (uint256) {
+        return bondingCheckpoints().getTotalActiveStakeAt(_round);
+    }
+
+    /**
+     * @notice Returns the delegate that _account has chosen. This means the delegated transcoder address in case of
+     * delegators, and the account own address for transcoders (self-delegated).
+     */
+    function delegates(address _account) external view returns (address) {
```

Comment:
I don't really like it either but this actually comes from the OZ interface that we have to implement: https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/f58645db634073c2c9be827c6959ace2f3733804/contracts/governance/utils/IVotesUpgradeable.sol#L53 (this is inherited by `IERC5805Upgradeable`). `delegate` is already used for the function that actually changes the delegate, as a verb.

If you think it improves readability I can add the `override` modifier (`virtual` as well?) to all the functions here that implement those interface methods. Solidity apparently doesn't require it anymore tho it might make things clearer here.

The only new method we've added here apart from the interface is `delegatedAt`, as defined in the `IVotes` interface on `GovernorCountingOverridable.sol`.

Reply:
I don't think it actually matters - just nitpicking! I think it makes sense to keep as-is if the parent uses that